### PR TITLE
Add support for controlling wheels targets

### PR DESCRIPTION
**Pull request by Cadair**
This implements the functionality to specify any valid cibuildwheel build string as a target in the publish template. The only restriction is an operating system substring to be able to detect the OS to use.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
+import platform
 from Cython.Build import cythonize
 
 
 here = path.abspath(path.dirname(__file__))
 with open(path.join(here, 'README.rst'), encoding='utf-8') as f:
     long_description = f.read()
+
+ext_modules = cythonize("fftw.pyx") if platform.system() != "Windows" else cythonize("fib.pyx")
+print(ext_modules)
```
Comment (Cadair) [inline setup.py:13]:
```suggestion
```

File: docs/publish.rst [hunk 2]
```diff
@@ -106,18 +107,55 @@ The ``test_command`` parameter gives a command that will be run in a temporary
...
+    - wheels_cp3?-win32
+    - wheels_cp3?-win_amd64
+
+
+You can also combine these specifications with the simple OS specifications, for
+those simple builds the ``CIBW_BUILD`` environment variable will not be set
+allowing you to set the defaults as above.
+
+The only restriction on this is that the specification must contain one of
+``win``, ``macosx`` or ``manylinux`` to be able to select the operating system
```
Comment (astrofrog) [inline docs/publish.rst:155]: maybe it's enough to say ``macos`` and ``linux`` since that should be sufficient?

File: publish.yml [hunk 3]
```diff
@@ -1,46 +1,45 @@
...
 jobs:
 - ${{ each target in parameters.targets }}:
 
-  - ${{ if or(eq(target, 'wheels_macos'), eq(target, 'wheels_linux'), eq(target, 'wheels_windows')) }}:
+  - ${{ if startsWith(target, 'wheels_') }}:
 
-    - job: ${{ target }}
+    - job: ${{ replace(replace(replace(replace(replace(replace(target, '?', ''), '*', ''), '[', ''), ']', ''), '!', ''), '-', '_') }}
       dependsOn: ${{ parameters.dependsOn }}
       condition: succeeded()
```
Comment (astrofrog) [inline publish.yml:12]: Set a displayname to the original target?

File: publish.yml [hunk 4]
```diff
@@ -1,46 +1,45 @@
...
 
       variables:
         CIBW_TEST_COMMAND: ${{ parameters.test_command }}
         CIBW_TEST_EXTRAS: ${{ parameters.test_extras }}
+        OA_TARGET: ${{ target }}
+        job_name: ${{ replace(replace(replace(replace(replace(replace(target, '?', ''), '*', ''), '[', ''), ']', ''), '!', ''), '-', '_') }}
+        ${{ if or(eq(target, 'wheels_macos'), contains(target, 'macosx')) }}:
```
Comment (astrofrog) [inline publish.yml:19]: Why not just:

```
        ${{ if or(contains(target, 'macos')) }}:
```

Since that should be unambiguous? Same for other OSes.

**Review by astrofrog — CHANGES_REQUESTED** [review]
Works nicely! Just a couple of small comments below.

**Comment by Cadair** [comment]
@astrofrog I think I addressed all your points, do you want to restart the astropy PR to test it once more?

**Comment by astrofrog** [comment]
I've restarted the CI on the astropy PR

**Review by astrofrog — APPROVED** [review]
Works great, thanks!

**Pull request merged**